src/lib/storage/vector_compression/simd_bp128/simd_bp128_iterator.hpp

```diff
@@ -18,77 +19,46 @@ class SimdBp128Iterator : public BaseCompressedVectorIterator<SimdBp128Iterator>
   using Packing = SimdBp128Packing;
 
  public:
-  SimdBp128Iterator(const pmr_vector<uint128_t>* data, const size_t size, const size_t absolute_index = 0u);
+  explicit SimdBp128Iterator(std::unique_ptr<SimdBp128Decompressor>&& decompressor, const size_t absolute_index = 0u)
+      : _decompressor{std::move(decompressor)}, _absolute_index{absolute_index} {}
 
-  SimdBp128Iterator(const SimdBp128Iterator& other);
-  SimdBp128Iterator& operator=(const SimdBp128Iterator& other);
+  explicit SimdBp128Iterator(const SimdBp128Iterator& other)
+      : _decompressor{std::make_unique<SimdBp128Decompressor>(SimdBp128Decompressor(*other._decompressor))},
+        _absolute_index{other._absolute_index} {}
+
+  explicit SimdBp128Iterator(SimdBp128Iterator&& other)
+      : _decompressor{std::move(other._decompressor)}, _absolute_index{std::move(other._absolute_index)} {}
+
+  SimdBp128Iterator& operator=(const SimdBp128Iterator& other) {
```

Comment:
Move code to cpp?

Reply:
Is there any advantage of that? I thought the few lines do not make a difference.

Reply:
That's what everyone says. A line here, two lines there, and the build is slow again.

Reply:
I don't like reading code where 90% of the code is in the hpp and I need to look at the cpp for just a few boring lines.

But if you say it might make a difference or it's a matter of code cleanliness, I'll change it.